src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect, useCallback, useMemo } from 'react';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { Loader } from '../Loader';
+import { Table } from '../Table';
+
+export const PeopleTable: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getPeople()
+      .then(setPeople)
+      .catch(() => setErrorMessage('Something went wrong'))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const setParent = useCallback((arrayOfPeople: Person[]): Person[] => {
+    return arrayOfPeople.map(person => {
+      const mother = arrayOfPeople.find(p => p.name === person.motherName);
+      const father = arrayOfPeople.find(p => p.name === person.fatherName);
+
+      return {
+        ...person,
+        mother,
+        father,
+      };
+    });
+  }, []);
```

Comment:
The `setParent` function recalculates the parent for each person every time it is called, which can be inefficient if the list of people is large. Consider optimizing this by using a map or object to store the parents and look them up in constant time.